File: src/main/java/com/atsuishio/superbwarfare/init/ModCommandArguments.java
package com.atsuishio.superbwarfare.init;

import com.atsuishio.superbwarfare.command.LowerCamelCaseEnumArgument;
import net.minecraft.commands.synchronization.ArgumentTypeInfo;
import net.minecraft.commands.synchronization.ArgumentTypeInfos;
import net.minecraft.core.registries.BuiltInRegistries;

public class ModCommandArguments {

    @SuppressWarnings({"rawtypes", "unchecked"})
    private static final ArgumentTypeInfo<LowerCamelCaseEnumArgument<?>, ?> INFO =
            ArgumentTypeInfos.register(
                    BuiltInRegistries.COMMAND_ARGUMENT_TYPE,
                    "superbwarfare:lower_camel_case_enum",
                    (Class) LowerCamelCaseEnumArgument.class,
                    new LowerCamelCaseEnumArgument.Info()
            );

    public static void register() {

    }
}

File: src/main/kotlin/com/atsuishio/superbwarfare/init/ModCommandArguments.kt
package com.atsuishio.superbwarfare.init

import com.atsuishio.superbwarfare.Mod
import com.atsuishio.superbwarfare.command.LowerCamelCaseEnumArgument
import net.fabricmc.fabric.api.command.v2.ArgumentTypeRegistry

object ModCommandArguments {

    private var initialized = false

    @JvmStatic
    fun init() {
        if (initialized) return
        initialized = true

        ArgumentTypeRegistry.registerArgumentType(
            Mod.loc("lower_camel_case_enum"),
            LowerCamelCaseEnumArgument::class.java,
            LowerCamelCaseEnumArgument.Info()
        )
    }
}
